worlds/sc2/rules.py

```diff
@@ -219,6 +219,46 @@ def great_train_robbery_train_stopper(self, state: CollectionState) -> bool:
                 or state.has_any({item_names.VULTURE, item_names.LIBERATOR}, self.player)
             )
         )
+    
+    def zerg_great_train_robbery_train_stopper(self, state: CollectionState) -> bool:
+        """
+        Ability to deal with trains (moving target with a lot of HP)
+        :param state:
+        :return:
+        """
+        return (
+            self.morph_impaler_or_lurker(state)
+            or state.has_all({item_names.MUTALISK_SEVERING_GLAIVE, item_names.MUTALISK}, self.player)
+            or state.has(item_names.ABERRATION, self.player)
+            or state.has(item_names.MALIGNANT_CREEP, self.player)
+            or self.advanced_tactics
+            and (
+                state.has_all({item_names.ZERGLING_BANELING_ASPECT, item_names.BANELING_CORROSIVE_ACID}, self.player)
+                or state.has_all({item_names.ROACH, item_names.ROACH_GLIAL_RECONSTITUTION}, self.player)
+            )
+        )
+    
+    def protoss_great_train_robbery_train_stopper(self, state: CollectionState) -> bool:
+        """
+        Ability to deal with trains (moving target with a lot of HP)
+        :param state:
+        :return:
+        """
+        return (
+            state.has_any({item_names.ANNIHILATOR, item_names.INSTIGATOR, item_names.STALKER}, self.player)
+            or state.has_all({item_names.SLAYER, item_names.SLAYER_PHASE_BLINK}, self.player)
+            or state.has(item_names.MATRIX_OVERLOAD, self.player)
```

Comment:
matrix overload gives pylons a buff field, so it makes sense here, but it needs units to buff. This needs to be paired with something, or be advanced tactics only under the assumption buffing the rescues is enough.